### PR TITLE
discussion: Activity Feed and Notifications

### PR DESCRIPTION
**Pull request by haganbmj**
Just wanted to open up the conversation to see if there were opinions about the implementation of Activity Feeds, Notifications, etc.

Most of my experience stems from relational databases, so I'm probably a bit off on models here.

I'll throw some comments within the files, but I imagine Notifications have to follow a similar model to an ActivityFeed, with the challenge being how items are marked eligible for addition to that list.

Relates to #49, #48, and #374.

### CODE REVIEW

File: models/activity.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+let mongoose = require('mongoose');
+
+let activitySchema = mongoose.Schema({
+  actor: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+    required: false
```
Comment (haganbmj) [inline models/activity.js]: Unsure here if this is correct, there likely should be a consideration for system level activity items - like website blog posts or updates. Those could be tied to a system User though.

File: models/activity.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  timestamp: {
+    type: Date,
+    required: true
+  },
+  referenceId: {
+    type: String,
```
Comment (haganbmj) [inline models/activity.js]: This likely has to be generic to account for a variation of activity types.

Given a Deck object the activity item can infer the type of message based on that model.
Similarly, a Blog object contains enough information to build a different format of activity message.

File: models/activityfeed.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+let mongoose = require('mongoose');
+
+let activityFeedSchema = mongoose.Schema({
+  owner: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
```
Comment (haganbmj) [inline models/activityfeed.js]: Having a User object here likely makes this more of a NewsFeed, this goes back again to my non-relational database inexperience though, unsure if this reference id should exist on the parent or child or both.

File: models/activityfeed.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    required: true
+  },
+  timestamp: {
+    type: Date,
+    required: true
+  },
+  activityId: {
```
Comment (haganbmj) [inline models/activityfeed.js]: The alternative here of course is to have an array of Activity.

**Comment by dekkerglen** [comment]
Are you on the discord? Most dev talk occurs there. 
https://discord.gg/Hn39bCU

@ dekkaru once you join to gain access to contributor channels

**Pull request closed without merging**